components/camel-hdfs/src/main/java/org/apache/camel/component/hdfs/HdfsConsumer.java

```diff
@@ -127,53 +129,67 @@ public boolean accept(Path path) {
     }
 
     private int processFileStatuses(HdfsInfo info, FileStatus[] fileStatuses) {
-        final AtomicInteger messageCount = new AtomicInteger(0);
+        final AtomicInteger totalMessageCount = new AtomicInteger(0);
 
-        Arrays.stream(fileStatuses)
+        List<HdfsInputStream> hdfsFiles = Arrays.stream(fileStatuses)
                 .filter(status -> normalFileIsDirectoryHasSuccessFile(status, info))
                 .filter(this::hasMatchingOwner)
-                .map(this::createInputStream)
+                .map(this::asHdfsFile)
                 .filter(Objects::nonNull)
-                .forEach(hdfsInputStream -> {
-                    try {
-                        processHdfsInputStream(hdfsInputStream, messageCount, fileStatuses.length);
-                    } finally {
-                        IOHelper.close(hdfsInputStream, "input stream", log);
-                    }
-                });
+                .collect(Collectors.toList());
 
-        return messageCount.get();
+        log.info("Processing [{}] valid files out of [{}] available.", hdfsFiles.size(), fileStatuses.length);
+
+        for (int i = 0; i < hdfsFiles.size(); i++) {
```

Comment:
IMHO I'd tend to replace this with `forEach` as it can be readable and concise, unless there was another reason perhaps?

Reply:
it's not for each, so I can print the index ... I can also add an int index and increment it, but in the end it's just syntactic sugar. An index will still be created by the compiler.

Reply:
Well for me that is a personal flavour that I tend to prefer forEach over the classical for loop, however if you really need index to print, then is fine for me